lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -306,6 +319,32 @@ export class PublicClientApplication {
 
     // #region Silent Flow
 
+    async ssoSilent(request: AuthenticationParameters): Promise<TokenResponse> {
+        // block the reload if it occurred inside a hidden iframe
+        BrowserUtils.blockReloadInHiddenIframes();
+
+        const silentRequest: AuthenticationParameters = {
+            ...request,
+            prompt: PromptValue.NONE
```

Comment:
`clientId` should be set as the only scope, right?

Reply:
No the point of this function is to acquire a code silently, regardless of scope. That is why I believe ssoSilent isn't a great name and we should change it here.

Reply:
Oh, I was under the impression the only scenario where iframe would be needed for token acquisition was the SSO scenario. When else would an iframe be needed to acquire a token in Auth Code?

Reply:
Regardless, I would argue the APIs should be as follows:

- `acquireTokenSilent`: Uses network client for most use cases, iframe handler if needed.
- `ssoSilent`: Uses iframe handler to acquire/renew ID token (similar to v1)

Reply:
Point being, you call `acquireTokenSilent` whenever you need an access token, and it figures out which handler to use. You call `ssoSilent` when you want to SSO to an existing session. The fact both might use an iframe is an implementation detail and apps shouldn't be required to know the difference.